### ISSUE TITLE
Getting error when using a non-default distribution

### ISSUE THREAD

**Issue by sankaravadivel** [indygreg/PyOxidizer#267]
I am trying to use Python 3.8.3 distribution and I am getting the following error
```
testuser@test:~/workspace/terraform-mae$ pyoxidizer build install
resolving 1 targets
resolving target install
resolving target exe
resolving target dist
resolving Python distribution Local { local_path: "/home/testuser/workspace/python-build-standalone/dist/cpython-3.8.3-linux64-20200611T0529.tar.zst", sha256: "418fdc998613ecc7647c4ca4a5f55e6b2f15d485487fc220ac3d5e4aeb74c923" }
existing /home/testuser/workspace/terraform-mae/./build/python_distributions/cpython-3.8.3-linux64-20200611T0529.tar.zst passes SHA-256 integrity check
Python distribution available at /home/testuser/workspace/terraform-mae/./build/python_distributions/cpython-3.8.3-linux64-20200611T0529.tar.zst
reading data from Python distribution...
error[PYOXIDIZER_BUILD]: missing field `arch` at line 1630 column 1
  --> ./pyoxidizer.bzl:65:11
   |
65 |       exe = dist.to_python_executable(
   |  ___________^
66 | |         name="mae",
67 | |         config=python_config,
68 | |         resources_policy="prefer-in-memory-fallback-filesystem-relative:lib",
...  |
96 | |         include_test=False,
97 | |     )
   | |_____^ resolve_distribution()


error: missing field `arch` at line 1630 column 1

```
This is the code I am having in my pyoxidizer.bzl

```
def make_dist():
    #return default_python_distribution()
    return PythonDistribution("418fdc998613ecc7647c4ca4a5f55e6b2f15d485487fc220ac3d5e4aeb74c923", local_path="/home/testuser/workspace/python-build-standalone/dist/cpython-3.8.3-linux64-20200611T0529.tar.zst")

```
I am getting the same error with the URL option as well.

**Comment by indygreg** [comment]
The semantics of the `PYTHON.json` file in distributions has changed a bit in recent weeks. I suspect what's happening here is you are using an old build of PyOxidizer with a Python distribution produced with a newer version of https://github.com/indygreg/python-build-standalone.

If you want to use the Python 3.8 distributions available from the `python-build-standalone` project, you'll need to install the latest commit of PyOxidizer from the `main` branch. Specifically, you'll need commit d442588dbc91f6b9069399030dd3ceba8d646491 or later. Although I suspect you'll want to grab the latest commit on `main` since there were some regressions with initial support for Python 3.8.

Python 3.8 is the future and I'd like to release a new version of PyOxidizer by the end of June with support.

**Comment by sankaravadivel** [comment]
Thanks for your response. I upgraded my PyOxidizer to the latest commit and I was able to go past that error. But now I am getting a different error. Any help with this is greatly appreciated.

```
libpythonXY created
building with Rust 1.44.0
    Updating crates.io index
error: failed to select a version for the requirement `pyembed = "^0.8.0-pre"`
  candidate versions found which didn't match: 0.7.0, 0.6.0
  location searched: crates.io index
required by package `mae v0.1.0 (/tmp/pyoxidizer.c0xzuDQv93vC/mae)`
error[PYOXIDIZER_BUILD]: cargo build failed
   --> ./pyoxidizer.bzl:142:5
    |
142 |     files.add_python_resource(".", exe)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ add_python_resource


error: cargo build failed

```

**Comment by indygreg** [comment]
That version error from the last comment was fixed in version 0.8.

I don't see anything obviously actionable in this issue. So I'm going to close it. If you feel differently, please comment!
